java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/converter/AvroMessageConverter.java

```diff
@@ -0,0 +1,39 @@
+package com.cumulocity.generic.mqtt.client.converter;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SchemaSerializationException;
+
+/**
+ * A converter to serialize message object to byte array and vice versa.
+ */
+public interface AvroMessageConverter<T> {
+
+    /**
+     * Encode an object representing the message content into a byte array.
+     *
+     * @param message the message object
+     * @return a byte array with the serialized content
+     * @throws SchemaSerializationException if the serialization fails
+     */
+    byte[] encode(T message);
+
+    /**
```

Comment:
issue: the deserializer implementation is implicit. It is this object.
suggestion: Decode a byte array into an object of type \<code\>T\</code\>.